psa-crypto/src/types/key_derivation_operation.rs

```diff
@@ -0,0 +1,265 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! # PSA Operation types
+
+use super::algorithm::{Hash, KeyDerivation};
+use super::key::Id;
+#[cfg(feature = "interface")]
+use super::status::{Error, Result, Status};
+#[cfg(feature = "interface")]
+use core::convert::{From, TryFrom};
+
+/// Key derivation operation for deriving keys from existing sources
+#[derive(Debug, Clone)]
+pub struct KeyDerivationOperation<'a> {
+    /// Key derivation algorithm and inputs
+    pub alg: KeyDerivationWithInputs<'a>,
+    /// Maximum capacity of a key derivation operation
+    pub capacity: Option<usize>,
```

Comment:
In the specs I can read:

> For some algorithms, this step is mandatory because the output depends on the maximum capacity.

If we know for which ones this is mandatory, we can force it for them and leave it as `Option` for the other ones?

Reply:
I did look into doing exactly that but couldn't find any documentation that states for which algorithms `capacity` is mandatory 😞.